scipy/ndimage/filters.py

```diff
@@ -1066,8 +1066,7 @@ def _min_or_max_filter(input, size, footprint, structure, output, mode,
                 raise RuntimeError("no footprint provided")
             separable = True
         else:
-            footprint = numpy.asarray(footprint)
-            footprint = footprint.astype(bool)
+            footprint = numpy.asarray(footprint, dtype=bool)
```

Comment:
I'm not as familiar with ndimage, but I assume that `footprint` is never modified by the native `min_or_max_filter` code? Otherwise, this change could lead to an in-place modification bug.